### PR TITLE
Added possibility to deleting reports older than date

### PR DESCRIPTION
**Pull request by georgRusanov**
I want to delete old reports by schedules, so I added the possibility to do it

### CODE REVIEW

File: src/main/java/ru/iopump/qa/allure/service/JpaReportService.java [hunk 1]
```diff
@@ -106,6 +107,20 @@ public void internalDeleteByUUID(UUID uuid) throws IOException {
         return res;
     }
 
+    public Collection<ReportEntity> deleteAllOlderThanDate(LocalDateTime date) throws IOException {
```
Comment (kochetkov-ma) [inline src/main/java/ru/iopump/qa/allure/service/JpaReportService.java]: Create new method in `JpaReportRepository` and delegate  `deleteAllOlderThanDate`  from JpaReportService to `JpaReportRepository`

File: src/main/java/ru/iopump/qa/allure/entity/ReportEntity.java [hunk 2]
```diff
@@ -90,6 +90,10 @@ public boolean isFullUrl() {
         return checkUrl(url);
     }
 
+    public UUID getUuid() {
```
Comment (kochetkov-ma) [inline src/main/java/ru/iopump/qa/allure/entity/ReportEntity.java]: It's redundant because there is Lombok `@Data` annotation on the class

**Review by kochetkov-ma — COMMENTED** [review]
Something went wrong during gradle trask `pmdMain`

```
* What went wrong:
Execution failed for task ':pmdMain'.
> java.lang.NoClassDefFoundError: com/fasterxml/jackson/module/jaxb/JaxbAnnotationIntrospector
```

Previous build was successful. But now I dont see the reasons your failed build. It would be great that you figure out the strange behaviour

**Comment by georgRusanov** [comment]
> Something went wrong during gradle trask `pmdMain`
> 
> ```
> * What went wrong:
> Execution failed for task ':pmdMain'.
> > java.lang.NoClassDefFoundError: com/fasterxml/jackson/module/jaxb/JaxbAnnotationIntrospector
> ```
> 
> Previous build was successful. But now I dont see the reasons your failed build. It would be great that you figure out the strange behaviour

@kochetkov-ma done

File: src/main/java/ru/iopump/qa/allure/entity/ReportEntity.java (hunk 2 above)
Comment (georgRusanov) [reply src/main/java/ru/iopump/qa/allure/entity/ReportEntity.java]: @kochetkov-ma done

File: src/main/java/ru/iopump/qa/allure/service/JpaReportService.java (hunk 1 above)
Comment (georgRusanov) [reply src/main/java/ru/iopump/qa/allure/service/JpaReportService.java]: @kochetkov-ma done

**Pull request merged**